### PR TITLE
Do not rfc3986 escape JSON bodies

### PR DESCRIPTION
**Pull request by simov**
Example `{json:{}}` or `{body:{}, json:true}`
Related to https://github.com/request/request/pull/1315
Closes https://github.com/request/request/issues/1395

Also take a look at the [options section](https://github.com/request/request#requestoptions-callback) of the documentation.

### CODE REVIEW

**Comment by nylen** [comment]
The behavior looks correct.  I tried to clean up the test a little to get comfortable that we're doing the right thing here, can you pull in https://github.com/nylen/request/commit/29f98f31ed1c90a08786e7c8345eaf0bbd289877?

File: request.js [hunk 1]
```diff
@@ -1428,15 +1428,15 @@ Request.prototype.json = function (val) {
     if (self.body !== undefined) {
       if (!/^application\/x-www-form-urlencoded\b/.test(self.getHeader('content-type'))) {
         self.body = safeStringify(self.body)
+      } else {
+        self.body = rfc3986(self.body)
```
Comment (nylen) [inline request.js]: Can you swap this conditional around?  Also, I see we have a test for it (`body + header + json`) but in what situation  is this needed?

**Comment by simov** [comment]
Done. Are we good with this one (regarding your comment from above)?

**Comment by nylen** [comment]
Yeah, thanks for taking a look!

**Comment by nylen** [comment]
version `2.53.0` released.

**Pull request merged**